pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -770,7 +769,25 @@ func (m *kubeGenericRuntimeManager) killContainer(ctx context.Context, pod *v1.P
 	klog.V(2).InfoS("Killing container with a grace period", "pod", klog.KObj(pod), "podUID", pod.UID,
 		"containerName", containerName, "containerID", containerID.String(), "gracePeriod", gracePeriod)
 
-	err := m.runtimeService.StopContainer(ctx, containerID.ID, gracePeriod)
+	var err error
```

Comment:
This would be the first time we background an execution inside the sync loop.  I'm a bit concerned why this is necessary - i would expect context cancellation to also cause runtimeService.StopContainer to exit.  Are you saying it does not?

Reply:
I don't think it will, the subsequent flow code that I've looked at.
The runtimeService.StopContainer was already sent to the CRI-O via grpc when the context was canceled.
The ctx.Done here is just to allow the kubelet to send the request once more in the

Reply:
I don't think it will, the subsequent flow code that I've looked at.
The runtimeService.StopContainer was already sent to the CRI-O via grpc when the context was canceled.
The ctx.Done here is just to allow the kubelet to send the request once more in the

Reply:
Although the previous code has ctx.cancel, there is no way to make runtimeService.StopContainer not exit properly without ctx.done listening here. It will keep blocking here.